pkg/hoist/hoist_launchable.go

```diff
@@ -322,10 +322,16 @@ func (hl *Launchable) Install(verifier auth.ArtifactVerifier) error {
 	return err
 }
 
-// The version of the artifact is currently derived from the location, using
-// the naming scheme <the-app>_<unique-version-string>.tar.gz
+func (hl *Launchable) tempFilePath() string {
+	return fmt.Sprintf("%s_%s.tar.gz", hl.Id, hl.Version())
```

Comment:
actually that just puts me back to the same behavior as before, i'll put it back for now and we can change the temp file path as part of a future PR when url end stops being acceptable.